### PR TITLE
Add submit test results example with results

### LINKED ISSUE #656: submit api should probably accept json
The submit results API (documented at https://squad.readthedocs.io/en/latest/intro.html#result-submit-ref-label) is unnecessarily difficult to use with python's requests library for two reasons.

1. It seems to require multipart/form-data, which requests() implements with "files=".
2. It accepts a json-based testrun, but it throws the payload away.

The behavior can be demonstrated with the program below, which submits a test result using json, followed by the same data using multipart/form-data (files=).

The first submission shows up at https://staging-qa-reports.linaro.org/api/testruns/1114287/.
The second submission shows up at https://staging-qa-reports.linaro.org/api/testruns/1114288/.

It took me a while to figure this out, so I hope we can at least update the documentation with a requests() example, and investigate supporting the more native json-based payload.

```py
import json
import os
import requests
import time

url = "https://staging-qa-reports.linaro.org/api/submit/lkft/linux-stable-rc-5.5-oe/v5.5/i386"

metadata = {
    "job_id": f"{time.time()}",
    "make_kernelversion": "5.5.0",
}
log = "make output"
tests_file = {
    "build/build-arm64-gcc-9": "pass",  # or "fail"
}
headers = {
    "Auth-Token": os.environ["QA_TOKEN"],
}

result = requests.post(
    url,
    headers=headers,
    json={"metadata": metadata, "log": log, "tests_file": tests_file},
)

if not result.ok:
    print(f"Error submitting to qa-reports: {result.reason}: {result.text}")


metadata = {
    "job_id": f"{time.time()}",
    "make_kernelversion": "5.5.0",
}
result = requests.post(
    url,
    headers=headers,
    files={
        "tests": json.dumps(tests_file),
        "metadata": json.dumps(metadata),
        "log": json.dumps(log),
    },
)

if not result.ok:
    print(f"Error submitting to qa-reports: {result.reason}: {result.text}")
```

### PR DESCRIPTION
**Pull request by chaws**
This PR adds an example of submitting test results through the api using Python's requests library. Also, it fixes a bug that happened when a float value was passed as job_id.

Closes #656

### CODE REVIEW

**Comment by chaws** [comment]
Thanks for the feedback @danrue! I think having it more secure by default is a good practice as well, less chance of hard-coding stuff in.

I also changed the other examples that used `xxxx` elsewhere in docs.

File: doc/intro.rst [hunk 1]
```diff
@@ -98,6 +98,26 @@ Example with test data as regular ``POST`` parameters::
...
 
+Example with test data using Python's requests library:
+
+.. code:: python
+    import json
+    import requests
+
+    tests = json.dumps({"test1": "pass", "test2": "fail"})
+    metrics = json.dumps({"metric1": 21, "metric2": 4})
+    metadata = json.dumps({"foo": "bar", "baz": "qux"})
```
Comment (mwasilew) [inline doc/intro.rst]: I think metadata needs some keys we consider 'mandatory'

Comment (chaws) [reply doc/intro.rst]: @mwasilew I reviewed the code that handles metadata, and no fields are actually mandatory. The only auto-filling we do is regarding job_id: https://github.com/Linaro/squad/blob/master/squad/core/tasks/__init__.py#L146

What we do have is `SPECIAL_FIELDS`: https://github.com/Linaro/squad/blob/master/squad/core/tasks/__init__.py#L117, but I don't think it's necessary to add them all to the examples. What do you think?

**Comment by chaws** [comment]
@mwasilew I've rebased it with master.

File: doc/intro.rst (hunk 1 above)
Comment (mwasilew) [reply doc/intro.rst]: OK, let's not use special fields.

**Pull request merged**